src/java.desktop/share/classes/sun/awt/FontConfiguration.java

```diff
@@ -136,7 +138,8 @@ protected void setOsNameAndVersion() {
     }
 
     private void setEncoding() {
-        encoding = Charset.defaultCharset().name();
+        encoding = Charset.forName(System.getProperty("native.encoding"),
+                                   UTF_8).name();
```

Comment:
It is extremely misleading that this hugely impactful and wrong change is slipped into a bug about AIX.
I don't much care what you do with AIX, but revert this.

The  native.encoding is not something we want to use here.
We had not over-looked that JEP 400 changed this and it is fine.
You are effectively undoing JEP 400 and making fontconfig files be found and use the old encoding when we really want it to match the default encoding as per the fontconfig spec
https://docs.oracle.com/en/java/javase/20/intl/font-configuration-files.html
    Encoding - the canonical name of the default encoding, as provided by java.nio.charset.Charset.defaultCharset().name().

Any env. on which line-spacing changes is because that is what is set up to be in that locale+encoding  and if you want to change it, update the fontconfig file for that locale+encoding.

@naotoj - please comment.

Reply:
I agree with Phil here. The proposed fix would violate what's defined in the fontconfig spec. I would expect AIX's font config to correctly choose an appropriate font file for the UTF-8 environment.